pkg/crc/api/api.go

```diff
@@ -6,26 +6,32 @@ import (
 	"fmt"
 	"net"
 
-	"github.com/code-ready/crc/pkg/crc/machine"
-
 	"github.com/code-ready/crc/pkg/crc/logging"
 )
 
-func CreateAPIServer(socketPath string, newConfig newConfigFunc, client machine.Client) (CrcAPIServer, error) {
+func CreateAPIServer(socketPath string, config newConfigFunc, machine newMachineFunc) (CrcAPIServer, error) {
 	listener, err := net.Listen("unix", socketPath)
 	if err != nil {
 		logging.Error("Failed to create socket: ", err.Error())
 		return CrcAPIServer{}, err
 	}
-	return createAPIServerWithListener(listener, newConfig, newHandler(client))
+	return createAPIServerWithListener(listener, config, machine)
 }
 
-func createAPIServerWithListener(listener net.Listener, newConfig newConfigFunc, handler RequestHandler) (CrcAPIServer, error) {
+func createAPIServerWithListener(listener net.Listener, config newConfigFunc, machine newMachineFunc) (CrcAPIServer, error) {
 	apiServer := CrcAPIServer{
 		listener:               listener,
-		newConfig:              newConfig,
 		clusterOpsRequestsChan: make(chan clusterOpsRequest, 10),
-		handler:                handler,
+		handlerFactory: func() (RequestHandler, error) {
+			cfg, err := config()
```

Comment:
Won't this re-create the issue we faced earlier in https://github.com/code-ready/crc/issues/1557? we solved that by re-creating the config for each handler

Reply:
Handler and his fields are recreated for each request, so it should be fine.

Reply:
Ahh, okay in here: https://github.com/code-ready/crc/pull/1526/files#diff-1b652f0fd14d465febc80f3ed2ed4b0f9e5a5cce6ed35c06869acd0bda0d95c4R61
Thanks!